src/Build/BackEnd/Components/Communications/NodeProviderOutOfProcTaskHost.cs

```diff
@@ -388,9 +388,13 @@ internal static void ClearCachedTaskHostPaths()
         /// <summary>
         /// Given a TaskHostContext, returns the name of the executable we should be searching for.
         /// </summary>
-        internal static string GetTaskHostNameFromHostContext(TaskHostContext hostContext)
+        internal static string GetTaskHostNameFromHostContext(HandshakeOptions hostContext)
         {
-            if (hostContext == TaskHostContext.X64CLR4 || hostContext == TaskHostContext.X32CLR4)
+            ErrorUtilities.ThrowInternalErrorUnreachable((hostContext & HandshakeOptions.TaskHost) == HandshakeOptions.TaskHost);
```

Comment:
If the condition is _false_, it throws, so this throws if it is not a task host that is passed in. I modeled it after cases like `ErrorUtilities.VerifyThrow(bool condition, string unformattedMessage)` where the first line of the body is `if (!condition)`.

If I wanted to do it in a switch/case of if/else, I'd either have to start with checking whether it's a task host (i.e., this) or put it at the end with every condition before it verifying that it's a task host. Since it's an error condition rather than an expected case, I think separating it somehow is better, and this is cleaner than adding a condition to every other case.

Reply:
```suggestion
            ErrorUtilities.VerifyThrowInternalErrorUnreachable((hostContext & HandshakeOptions.TaskHost) == HandshakeOptions.TaskHost);
```